x-pack/plugins/reporting/server/lib/event_logger/logger.ts

```diff
@@ -27,171 +29,192 @@ export interface ExecutionCompleteMetrics {
   byteSize: number;
 }
 
+export interface IReportingEventLogger {
+  logEvent(message: string, properties: LogMeta): void;
+  startTiming(): void;
+  stopTiming(): void;
+}
+
+const getEventLog = (logger: LevelLogger) => ({
```

Comment:
I am wondering why don't we reuse the Adapter pattern here? It would look less confusing.

```typescript
class EventLog implements IReportingEventLogger {
  constructor(private logger: LevelLogger, private properties: Partial<LogMeta>) {}

  // ...
}
```

Reply:
I had to stick to the rule of "one class per file." To make it less confusing, I moved this out to a separate file and implemented it as a class: 353794475a3